### PR TITLE
bot: label PRs based on their size

### PR DESCRIPTION
**Pull request by zmb3**
This does not take generated files into account.

Fixes gravitational/teleport#14214

### CODE REVIEW

File: bot/internal/bot/bot.go [hunk 1]
```diff
@@ -146,7 +159,17 @@ func isLargePR(files []github.PullRequestFile) bool {
 		additions += f.Additions
 		deletions += f.Deletions
 	}
-	return additions-deletions > 1500
+	delta := additions - deletions
+	switch {
+	case delta < 500:
```
Comment (r0mant) [inline bot/internal/bot/bot.go:164]: Nit: Maybe < 100 for small? It just "feels" like anything over 100 characters is not a trivial changeset.

File: bot/internal/bot/bot.go [hunk 2]
```diff
@@ -137,7 +137,20 @@ func classifyChanges(e *env.Environment, files []github.PullRequestFile) (docs b
...
+	return prSize(files) == xlarge
+}
+
+type sizeLabel string
+
+const (
+	small  sizeLabel = "size-sm"
+	medium sizeLabel = "size-md"
+	large  sizeLabel = "size-lg"
+	xlarge sizeLabel = "size-xl"
```
Comment (r0mant) [inline bot/internal/bot/bot.go:150]: Nit: What do you think about "namespacing" sizes like we do for other labels like "db/", etc.

```suggestion
	small  sizeLabel = "size/sm"
	medium sizeLabel = "size/md"
	large  sizeLabel = "size/lg"
	xlarge sizeLabel = "size/xl"
```

**Review by r0mant — APPROVED** [review]
lgtm with a couple of suggestions.

File: bot/internal/bot/bot.go (hunk 2 above)
Comment (zmb3) [reply bot/internal/bot/bot.go:150]: Sure!

File: bot/internal/bot/bot.go (hunk 1 above)
Comment (zmb3) [reply bot/internal/bot/bot.go:164]: Sure, totally open to changing the sizes. (BTW, these are _lines_, not characters)

My goal for this was more to convey to a reviewer when they open a PR how much they have to review. IMO, "small" should mean "I can review this in 10 minutes or less." Do you think 100 is the right number for that?

Comment (r0mant) [reply bot/internal/bot/bot.go:164]: I think it depends on the reviewer tbh. For me personally, 500 lines feels like something I should carve out time for and not an easy fix. 100 feels like something that can be reviewed quickly. We can always experiment and tweak the numbers.

**Pull request merged**